packages/core-mobile/app/services/swap/SwapService.ts

```diff
@@ -141,14 +133,19 @@ class SwapService {
         if (!SUPPORTED_SWAP_NETWORKS.includes(network.chainId)) {
           throw new Error(`${network.chainName} is not supported by Paraswap`)
         }
-        const fetcher = constructFetchFetcher(fetch)
-        const { getSpender } = constructGetSpender({
-          apiURL: this.getParaSwapSDK(network.chainId).apiURL,
-          chainId: ChainId.AVALANCHE_MAINNET_ID,
-          version: ParaSwapVersion.V6,
-          fetcher
-        })
-        return getSpender()
+        try {
+          const fetcher = constructFetchFetcher(fetch)
+          const { getSpender } = constructGetSpender({
+            apiURL: this.getParaSwapSDK(network.chainId).apiURL,
+            chainId: ChainId.AVALANCHE_MAINNET_ID,
+            version: ParaSwapVersion.V6,
+            fetcher
+          })
+          return getSpender()
+        } catch (error) {
+          Logger.error('Failed to get Paraswap spender', error)
+          return undefined
```

Comment:
shouldn't we just throw an error here?

Reply:
<img width="701" alt="Screenshot 2024-08-29 at 12 33 59 PM" src="https://github.com/user-attachments/assets/5963d4c9-e813-43c2-b6c2-781bc5a49075">

if spenderAddress is undefined, it will failed to get the allowance, and throw `allowance error`

Reply:
I think it makes more sense to throw an error instead of returning undefined. The `Allowance Error` represents a different error path, and throwing an error aligns naturally with the existing pattern in the `getParaswapSpender` function, where other errors are already being thrown.​

Reply:
update [here](https://github.com/ava-labs/core-mobile/pull/1527/commits/c5d3c2b37dc9de8a76d33a883f84902890c575f3)